core/embed/rust/src/trezorhal/display.rs

```diff
@@ -74,20 +104,28 @@ pub fn image(x: i32, y: i32, w: i32, h: i32, data: &[u8]) {
 pub fn toif_info(data: &[u8]) -> Result<ToifInfo, ()> {
     let mut width: cty::uint16_t = 0;
     let mut height: cty::uint16_t = 0;
-    let mut grayscale: bool = false;
+    let mut format: ffi::toif_format_t = ffi::toif_format_t_TOIF_FULL_COLOR_BE;
     if unsafe {
         ffi::display_toif_info(
             data.as_ptr() as _,
             data.len() as _,
             &mut width,
             &mut height,
-            &mut grayscale,
+            &mut format,
         )
     } {
+        let format = match format {
+            ffi::toif_format_t_TOIF_FULL_COLOR_BE => ToifFormat::FullColorBE,
```

Comment:
set explicit values from the original enum, i.e., `FullColorBE = ffi::toif_format_t_TOIF_FULL_COLOR_BE`

then you can use `num` and `num_derive` to `derive(FromPrimitive)` and use `let format = ToifFormat::from_usize(format as usize).unwrap_or(ToifFormat::FullColorBE)`

(see the relevant crates here: https://github.com/trezor/trezor-firmware/commit/f6c01d3dd0d405b7070d7c7104f5890fb290de11#diff-178b940e921da76b640eaf8e8d0214ec45db8bb824713299c9dbc743fbbdd84aR46-R52)

Reply:
done https://github.com/trezor/trezor-firmware/commit/610abe5338c4a9e9be87ce90557763042f01214e